src/lib/schema.ts

```diff
@@ -88,15 +99,58 @@ class MddbFile {
     if (!areUniqueObjectsByKey(files, "file_path")) {
       throw new Error("Files must have unique file_path");
     }
+
     const serializedFiles = files.map((file) => {
-      return {
-        ...file,
-        metadata: JSON.stringify(file.metadata),
-      };
+      const serializedFile: any = {};
+      const defaultProperties = [
+        "_id",
+        "file_path",
+        "extension",
+        "url_path",
+        "filetype",
+      ];
+
+      for (const key in file) {
+        if (Object.prototype.hasOwnProperty.call(file, key)) {
+          const value = file[key];
+          // If the value is undefined, default it to null
+          serializedFile[key] = defaultProperties.includes(key)
+            ? value
+            : value !== undefined
+            ? JSON.stringify(value)
+            : null;
+        }
```

Comment:
The comment is misleading. What are you trying to achieve here? There must be a better way to code this...